app/portainer/views/account/CreateAccessToken/CreateAccessToken.tsx

```diff
@@ -0,0 +1,107 @@
+import { PropsWithChildren, useState } from 'react';
+
+import { FormControl } from '@/portainer/components/form-components/FormControl';
+import { TextInput } from '@/portainer/components/form-components/Input';
+import { Button } from '@/portainer/components/Button';
+import { Heading } from '@/portainer/components/form-components/Heading';
+import { TextTip } from '@/portainer/components/Tip/TextTip';
+import { Code } from '@/portainer/components/Code';
+import { CopyButton } from '@/portainer/components/Button/CopyButton';
+
+import styles from './CreateAccessToken.module.css';
+
+interface AccessTokenResponse {
+  rawAPIKey: string;
+}
+
+export interface Props {
+  // userId for whom the access token is generated for
+  userId: number;
+
+  // onSubmit dispatches a successful matomo analytics event
+  onSubmit: (description: string) => Promise<AccessTokenResponse>;
+
+  // onSuccess is called when upon clicking the done button
+  onSuccess: () => void;
+
+  // onError is called when an error occurs; this is a callback to Notifications.error
+  onError: (heading: string, err: unknown, message: string) => void;
+}
+
+export function CreateAccessToken({
+  onSubmit,
+  onSuccess,
+  onError,
+}: PropsWithChildren<Props>) {
+  const [description, setDescription] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [accessToken, setAccessToken] = useState('');
+
+  // TODO: use axios or API service
+  async function generateAccessToken() {
+    if (isLoading) {
+      return;
+    }
+
+    setIsLoading(true);
+    try {
+      const response = await onSubmit(description);
+      setAccessToken(response.rawAPIKey);
+    } catch (err) {
+      onError('Failure', err, 'Failed to generate access token');
+    } finally {
+      setIsLoading(false);
+    }
+  }
+
+  function getError() {
+    if (description.length === 0) {
```

Comment:
I like it. did you have a look at formik?

Reply:
something like useState for errors, and useEffect that is called whenever description is changed might be more optimal

Reply:
Haven't looked at formik; good point on useEffect!